devtools/analyze_diag.py

```
@@ -0,0 +1,682 @@
+    if optimizer:
+        stop_w = 2.0
+        pt = optimizer.analyze_power_thresholds()
+        if pt:
+            stop_w = pt.get("suggested_stop_threshold_w", 2.0)
+            suggestions[CONF_STOP_THRESHOLD_W] = {
+                "value": round(stop_w, 2),
+                "reason": f"5th-percentile of minimum-active-power across {len(labeled)} labeled cycles.",
+            }
+            suggestions[CONF_START_THRESHOLD_W] = {
+                "value": round(pt.get("suggested_start_threshold_w", stop_w * 1.5), 2),
+                "reason": "Hysteresis band above the stop threshold (×1.2).",
```

Ruff (0.15.10)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)